TechnicalDocuments/Proposals/IdentifiersForUIAutomation.md

```diff
@@ -0,0 +1,203 @@
+# Improve discoverability of elements in UI automation tests
+
+* Author(s): Anders Ha
+* Review Manager: Jason Dobo
+
+## Introduction
+At the moment, there is no systematic approach to apply accessibility identifiers in Bento. Some components accept a custom accessibility identifier as a temporary measure, e.g. Chatbot UI components.
+
+## Motivation
+Use of ad hoc `String` and textual content as accessibility identifiers makes automation tests fragile, and there were already a few breakages observed due to mismatch between the UI code and the automation test code. This is because the main codebase does not publish accessibility identifiers used, and therefore automation tests have no mean to ensure themselves always pick up the same identifier as defined in the main codebase. It also raises the barrier of running automation tests on different locales.
+
+## Proposed solution
+
+### New Bento mechanism
+Bento shall introduce a new requirement to `ViewLifecycleAware`:
+
+```swift
+public protocol ViewLifecycleAware: NativeView {
+    // ... existing requirements.
+
+    func updateAccessibilityIdentifiers(usingPrefix prefix: String)
+}
+```
+
+When Bento is built with a Debug configuration, it should invoke `updateAccessibilityIdentifiers(usingPrefix:)` on the component root view, whenever it is bound to a given component at a given ID path for the first time.
+
+Components are free to set accessibility identifiers in whatever way they see fit, but they *must* append the given prefix to any accessibility identifier it set.
+
+Bento shall generates a prefix based on the section ID and the item ID. All variables are suffixed by `/`. So for example, the prefix for item `card` of section `activity` should be `activity/card/`.
```

Comment:
> Bento shall generates a prefix based on the section ID and the item ID

A lot of times these ids contain unique identifiers of the DTOs they are rendering. Considering that this can be UUIDs (and even with simpler Ints) this will make such generated ids hard to use in practice

Reply:
If possible we should not use UUIDs are part of identifiers

Reply:
If we have to be free from these payloads for uniqueness purpose, we would need sections and nodes to define an automation name/ID that is free of these matters.

Reply:
However, if we go along this way, we might as well just have components accepting accessibility IDs.